### DISCUSSION TITLE
All FieldsResolver in my project returning null

### DISCUSSION THREAD

**Discussion by igorwessel** [MichalLytek/type-graphql#997] [Q&A]
Hello!
My english is very bad, so in advance sorry for english =D

I don't why FieldsResolver is returning null because i don't update anything in project
`  @FieldResolver(() => Tools, { nullable: true })
   async tools(@Root() ninja: Ninja, @Ctx() { loaders: { ninjaToolLoader } }: IGraphQLContext): Promise<Tools[]> {
   const tools: Tools[] = await ninjaToolLoader.load(ninja.id);
   return tools;
}
`

When I do a test with a request return null
![Captura de tela de 2021-08-27 17-14-33](https://user-images.githubusercontent.com/54173994/131183950-3d2daa0e-7998-44a8-8d7b-13123f811b16.png)

I spent the whole morning and afternoon trying to solve it but I couldn't :/

**Comment by MichalLytek** [comment]
Have you tried returning dummy data directly, skipping the loader/database code?

**Reply by igorwessel** [reply]
Yeah i tried but is the same result, my dataloader is returning data correct
![Caputa de tela 2021-07-27 15-55](https://user-images.githubusercontent.com/54173994/131183719-6928b580-8861-4c32-a25c-774db16fd0b9.png)

I have the github repository is naruto_api in my repo

**Reply by MichalLytek** [reply]
It should be `@FieldResolver(() => [Tools]`. Use `[Brackets]` to mark field as returning array.

**Reply by MichalLytek** [reply]
You should return `error` apart from null in `data` from nullable field.

**Reply by igorwessel** [reply]
Ok! I was returning an empty array when I had no tools, I will try to figure out how to return an error apart.
And i changed the code to 

	@FieldResolver(of => [Tools])

	async tools(@Root() ninja: Ninja, @Ctx() { loaders: { ninjaToolLoader } }: IGraphQLContext): Promise<Tools[]> {

		const tools: Tools[] = await ninjaToolLoader.load(ninja.id);
		return tools;
	}

But don't work yet, it was working, I don't know if there was any dependency that updated and ended up causing this strange problem

**Reply by MichalLytek** [reply]
You can also try to do `@Resolver(() => Ninja)`.
If you want me to help, create a minimal reproducible code example, without dataloaders, database and other deps. One resolver, one class, simple setup.

**Reply by igorwessel** [reply]
thanks for trying to help! 
I redid my repository from what I had in my github, installed the dependencies putting to not update. And it worked!
I'm going to try to find out now which updated dependency caused to break FieldsResolvers
